pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -542,10 +542,15 @@ def _binary_op_method_timedeltalike(op, name):
 
         elif hasattr(other, 'dtype'):
             # nd-array like
-            if other.dtype.kind not in ['m', 'M']:
-                # raise rathering than letting numpy return wrong answer
+            try:
+                if other.dtype.kind in ['m', 'M']:
+                    return op(self.to_timedelta64(), other)
+                elif other.dtype.kind == 'O':
+                    return np.array([op(self, x) for x in other])
+                else:
+                    return NotImplemented
+            except:
```

Comment:
What does the try/except catch?

Reply:
That explains the `return NotImplemented` on 551, but not the try/except on 552.  I'm pretty sure the try/except is not needed.

Reply:
well if calling the operation on 549 throws an exception wouldn't you want it to not error and instead return a proper message? that was why i caught any issue there. object types could be more than just timestamps, etc.

Reply:
> well if calling the operation on 549 throws an exception wouldn't you want it to not error 

I'd want the traceback to show where the actual error came from.  Catching and re-raising obscures that.

Reply:
Done.